integration/aggregate_compat_test.go

```diff
@@ -345,7 +345,12 @@ func TestAggregateCompatSort(t *testing.T) {
 				{"v.foo", 1},
 				{"_id", 1}, // sort by _id when v is the same.
 			}}}},
-			skip: "https://github.com/FerretDB/FerretDB/issues/2101",
```

Comment:
I added a few more tests in addition to this one. So, now it covers a bit more cases related to dot notation. If you see some other examples that need to be covered, don't hesitate to propose them.

Also, I see Tigris started failing with timeout. I didn't add too many cases, so I'm not sure why. Maybe someone has an idea how to handle it (or should I add one more Tigris instance maybe?)

Reply:
@chilagrow hmm, do you have a link where it shows which errors it returns on CI? It passes locally, so if it fails on CI, it's something interesting. I see that CI returns a timeout from time to time for Postgres, and I wonder why, as I didn't add too many tests.

Reply:
See this one https://github.com/FerretDB/FerretDB/actions/runs/4372348012/jobs/7649165466
It failed on 3 tests and one of which is this test.

```
2023-03-09T08:21:13.1577454Z     --- FAIL: TestQueryCompatRunner/ImplicitEqBoolTrue (0.37s)
2023-03-09T08:21:13.1577916Z         --- FAIL: TestQueryCompatRunner/ImplicitEqBoolTrue/TestQueryCompatRunner_Bools (0.02s)
2023-03-09T08:21:13.1578383Z             query_compat_test.go:193: Compat (expected) IDs: [bool-true]
2023-03-09T08:21:13.1578709Z             query_compat_test.go:193: Target (actual)   IDs: []
2023-03-09T08:21:13.1578942Z             query_compat_test.go:193: 
2023-03-09T08:21:13.1579512Z                    Error Trace:    /home/runner/work/FerretDB/FerretDB/internal/util/testutil/equal.go:66
2023-03-09T08:21:13.1580310Z                                                            /home/runner/work/FerretDB/FerretDB/integration/helpers.go:152
2023-03-09T08:21:13.1581205Z                                                            /home/runner/work/FerretDB/FerretDB/integration/query_compat_test.go:126
2023-03-09T08:21:13.1581440Z                    Error:          Not equal: 
2023-03-09T08:21:13.1581720Z                                    expected: [
2023-03-09T08:21:13.1581942Z                                      {
2023-03-09T08:21:13.1582224Z                                        "$k": [
2023-03-09T08:21:13.1582519Z                                          "_id",
2023-03-09T08:21:13.1582797Z                                          "v"
2023-03-09T08:21:13.1583040Z                                        ],
2023-03-09T08:21:13.1583468Z                                        "_id": "bool-true",
2023-03-09T08:21:13.1583769Z                                        "v": true
2023-03-09T08:21:13.1583988Z                                      }
2023-03-09T08:21:13.1584190Z                                    ]
2023-03-09T08:21:13.1584477Z                                    actual  : []
2023-03-09T08:21:13.1584797Z                                    --- expected
2023-03-09T08:21:13.1585065Z                                    +++ actual
2023-03-09T08:21:13.1585408Z                                    @@ -1,10 +1 @@
2023-03-09T08:21:13.1585641Z                                    -[
2023-03-09T08:21:13.1585901Z                                    -  {
2023-03-09T08:21:13.1586221Z                                    -    "$k": [
2023-03-09T08:21:13.1586557Z                                    -      "_id",
2023-03-09T08:21:13.1586864Z                                    -      "v"
2023-03-09T08:21:13.1587132Z                                    -    ],
2023-03-09T08:21:13.1587558Z                                    -    "_id": "bool-true",
2023-03-09T08:21:13.1587901Z                                    -    "v": true
2023-03-09T08:21:13.1588163Z                                    -  }
2023-03-09T08:21:13.1588389Z                                    -]
2023-03-09T08:21:13.1588605Z                                    +[]

```